esmvalcore/cmor/table.py

```diff
@@ -122,7 +122,14 @@ def _assign_dimensions(self, var, generic_levels):
                 coord.generic_level = True
                 coord.axis = 'Z'
             else:
-                coord = self.coords[dimension]
+                try:
+                    coord = self.coords[dimension]
+                except KeyError:
+                    logger.exception(
+                        'Can not find dimension %s for variable %s',
```

Comment:
It would be nice if you could also mention the mip that the variable is defined in

Reply:
I do not have that info in this method and seems quite ugly to pass extra variables just for error messages

Reply:
Now all tables log the exact table where the error happens if an exception is raised and then raise again.
I think is better than logging that at this stage, as will benefit all possible errors fired when reading the tables

Reply:
Now all tables log the exact table where the error happens if an exception is raised and then raise again.
I think is better than logging that at this stage, as will benefit all possible errors fired when reading the tables